src/operator/regression_output-inl.h

```diff
@@ -69,11 +69,11 @@ class RegressionOutputOp : public Operator {
     Stream<xpu> *s = ctx.get_stream<xpu>();
     real_t num_output =
       in_data[reg_enum::kLabel].Size()/in_data[reg_enum::kLabel].shape_[0];
-    Tensor<xpu, 2> out = out_data[reg_enum::kOut].FlatTo2D<xpu, real_t>(s);
-    Tensor<xpu, 2> grad = in_grad[reg_enum::kData].FlatTo2D<xpu, real_t>(s);
-    Tensor<xpu, 2> label = in_data[reg_enum::kLabel]
-      .get_with_shape<xpu, 2, real_t>(out.shape_, s);
-    Assign(grad, req[reg_enum::kData], param_.grad_scale/num_output*
+    Tensor<xpu, 2, DType> out = out_data[reg_enum::kOut].FlatTo2D<xpu, DType>(s);
+    Tensor<xpu, 2, DType> grad = in_grad[reg_enum::kData].FlatTo2D<xpu, DType>(s);
+    Tensor<xpu, 2, DType> label = in_data[reg_enum::kLabel]
+      .get_with_shape<xpu, 2, DType>(out.shape_, s);
+    Assign(grad, req[reg_enum::kData], scalar<DType>(param_.grad_scale/num_output)*
```

Comment:
yep... I'll propose a pr to fix nullptr issue in other operators later. Is there any paper list to share about uint8 network?

There are some other issues to be fixed to make DType network really works, like kvstore, param init. For the lstm case, extra states data_type is need.  We need to discuss more about this DType support and a more detail todo list.